packages/app/src/components/nominationInfo/ApplicationForm.js

```diff
@@ -0,0 +1,311 @@
+import React, { useEffect, useRef, useContext, useState } from 'react';
+// import styles from './styles.module.css';
+import styles from './newstyles.module.css';
+import { useForm, Controller } from 'react-hook-form';
+import { yupResolver } from '@hookform/resolvers/yup';
+import * as Yup from 'yup';
+import nominationsAPI from '../../utils/API/nominationsAPI';
+import { NominationsDataContext } from '../../utils/context/NominationsContext';
+import { ActiveNominationContext } from '../../utils/context/ActiveNominationContext';
+import DatePicker, { CalendarContainer } from 'react-datepicker';
+import 'react-datepicker/dist/react-datepicker.css';
+
+const ApplicationForm = props => {
+  const firstUpdate = useRef(true);
+
+  const [NominationsData, setNominationsData] = useContext(
+    NominationsDataContext
+  );
+
+  const [activeNomination, setActiveNomination] = useContext(
+    ActiveNominationContext
+  );
+
+  const [admissionDate, setAdmissionDate] = useState(new Date());
+  const [dischargeDate, setDischargeDate] = useState(new Date());
+
+  useEffect(() => {
+    if (props.formData['Admission Date'] != 'Invalid Date') {
+      setAdmissionDate(props.formData['Admission Date']);
+    }
+  }, [props.formData['Admission Date']]);
+
+  useEffect(() => {
+    if (props.formData['Discharge Date'] != 'Invalid Date') {
+      setDischargeDate(props.formData['Discharge Date']);
+    }
+  }, [props.formData['Discharge Date']]);
+
+  const phoneRegex = /^[\+]?[(]?[0-9]{3}[)]?[-\s\.]?[0-9]{3}[-\s\.]?[0-9]{4,6}$/;
+  const yesNoRegex = /^(?:Yes\b|No\b)/;
+
+  useEffect(() => {
+    if (!firstUpdate.current) {
+      handleSubmit(submitForm)();
+    }
+    firstUpdate.current = false;
+  }, [props.saveHasBeenClicked]);
+
+  const validationSchema = Yup.object({
+    'Admission Date': Yup.date().required('Required'),
+    'Discharge Date': Yup.date()
+      .min(
+        Yup.ref('Admission Date'),
+        'Discharge date cannot be before admission date.'
+      )
+      .required('Required'),
+    'Representative Name': Yup.string()
+      .min(3, 'Must be 3 characters or more.')
+      .max(30, 'Must be 30 characters or less.')
+      .required('Required'),
+    'Representative Email Address': Yup.string()
+      .email('Invalid email address.')
+      .required('Required'), // This handles email validation with no regex.
+    'Representative Phone Number': Yup.string()
+      .matches(phoneRegex, 'Please enter a valid phone number.')
+      .required('Required'),
+    Relationship: Yup.string()
+      .min(3, 'Must be at least 3 characters.')
+      .max(20, 'Must be no more than 20 characters.')
+      .required('Required'),
+    'Request to communicate in Spanish?': Yup.string()
```

Comment:
Instead of making the user type in 'yes' or 'no', I'd probably make this a checkbox so there's no chance of invalid input and the user has to do less work.

Reply:
the user does not type yes or no now. it is a <select> element thingamajigger